app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -2,10 +2,34 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   include PreferencesController
 
   require_authentication :all, scopes: [:project]
+  extra_schema_actions :update_settings
   schema_type :json_schema
+  before_action :validate_records, only: [:update_settings]
+
+  def update_settings
+    upp = UserProjectPreference.find_or_initialize_by(project_id: params_for[:project_id],
```

Comment:
I think the intention is revealed in the name of the method on the resource, `update`. Just build this update out for now as it's the use case you have for Gravity Spy